subprojects/docs/src/docs/userguide/jvm/jvm_test_suite_plugin.adoc

```diff
@@ -0,0 +1,162 @@
+// Copyright 2021 the original author or authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+[[jvm_test_suite_plugin]]
+= The JVM Test Suite Plugin
+
+The JVM Test Suite plugin (plugin id: `jvm-test-suite`) allows for defining multiple groups of automated tests in a Java project.  These tests can be compiled once and run multiple times against various different target environments (for example, on different JVMs).
+
+This plugin can be used to define a group of Integration Tests, which might run much longer than unit tests and have different environmental requirements, and ought to be run against all the different JVMs which will be supported in production.
+
+[[sec:jvm_test_suite_usage]]
+== Usage
+
+This plugin is applied automatically by the `java` plugin, but can be explicitly applied if desired:
+
+.Using the JVM Test Suite plugin
+====
+include::sample[dir="snippets/testing/test-suite-plugin/groovy",files="build.gradle[tags=apply-jvm-test-suite]"]
+include::sample[dir="snippets/testing/test-suite-plugin/kotlin",files="build.gradle.kts[tags=apply-jvm-test-suite]"]
+====
+
+The plugin adds a `testing` extension (type: link:{groovyDslPath}/org.gradle.testing.base.TestingExtension.html[TestingExtension]) to the project used to configure test suites.
+
+[[sec:jvm_test_suite_tasks]]
+== Tasks
+
+The JVM Test Suite plugin adds the following task to the project:
+
+`test` — link:{groovyDslPath}/org.gradle.api.tasks.testing.Test.html[Test]::
+_Depends on_: `testClasses` from the `java` plugin, and all tasks which produce the test runtime classpath
++
+Runs the unit tests using JUnit by default; can be configured for other frameworks.
+
+Additional instances of link:{groovyDslPath}/org.gradle.api.tasks.testing.Test.html[Test] tasks will be automatically created for each test suite added via the `testing` extension.
+
+[[sec:jvm_test_suite_configuration]]
+== Configuration
+
+See the link:{groovyDslPath}/org.gradle.testing.base.TestingExtension.html[TestingExtension] class in the API documentation.
+
+[[sec:jvm_test_suite_terminology]]
+=== Terminology and Modeling
+The JVM Test Suite Plugin introduces some modeling concepts backed by new APIs.  Here are their definitions.
+
+==== Test Suite ====
+Each test suite is a _component_. A component is an abstract software thing. It is the description of the software to build.
```

Comment:
I think we don't need to include these details about a component just yet. This definition was more for internal use and not a user. 
To steal some inspiration from the Javadoc for JvmTestSuite:
> A test suite is a collection of JVM-based tests.